spec/userLogInSpec.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+describe("user log in tests", () => {
+  let userLogIn;
+  let response;
+
+  it("successfully logs in", () => {
+    let handle = "JasmineTest";
+    let password = "password";
+    response = new Response(
+      JSON.stringify({
+        user_id: 999,
+        session_key: "_2a_12_Pgo1hhpBTcmVE9_oqFHyNu",
+      })
+    );
+    spyOn(window, "fetch").and.returnValue(Promise.resolve(response));
+    userLogIn = new UserLogIn();
+    return userLogIn.logIn(handle, password).then(() => {
```

Comment:
Why do you return here?

Reply:
I'm not sure to be honest, I seem to do it in all the tests where I'm spying on fetch. I'll have a closer look at how this test works (I cobbled together the tests from bits and pieces of information I found from googling).